mixnet/config.py

```diff
@@ -2,110 +2,56 @@
 
 import random
 from dataclasses import dataclass
-from typing import List, TypeAlias
+from typing import List
 
 from cryptography.hazmat.primitives.asymmetric.x25519 import (
     X25519PrivateKey,
     X25519PublicKey,
 )
-from pysphinx.node import Node
-
-from mixnet.bls import BlsPrivateKey, BlsPublicKey
-from mixnet.fisheryates import FisherYates
+from pysphinx.sphinx import Node as SphinxNode
 
 
 @dataclass
 class MixnetConfig:
-    topology_config: MixnetTopologyConfig
-    mixclient_config: MixClientConfig
-    mixnode_config: MixNodeConfig
+    node_configs: List[NodeConfig]
+    membership: MixMembership
 
 
 @dataclass
-class MixnetTopologyConfig:
-    mixnode_candidates: List[MixNodeInfo]
-    size: MixnetTopologySize
-    entropy: bytes
+class NodeConfig:
```

Comment:
I think the max number of connections is the part of underlying gossip protocol but should be set when running the node. The preferred and maximum length of k-path should be a node parameter. 
I think it's meaningful to specify the k-path length parameter in this spec. I'll do it as a next PR. Especially the max length is important for limiting the size of Sphinx packet as a constant regardless of the actual k-path length that each node will choose. I'll clarify it in this spec in the next PR.
Also, the number of connections is important for defining the proper GTR (Global Transmission Rate) because it's related to the max bandwidth capacity of each node. I'm not sure if we need to specify it in this spec, but I'll define/use this param for simulation.

Reply:
This is a good point. Thank you. I added a commit 3585d3c which clarify what's a part of global config and what's a local node config.

Reply:
I think that number of connections and GTR are important from the perspective of simulation. These two parameters (alongside the queueing mechanism) will define the speed at which messages are propagated through the network. This might not be important from the spec perspective, but if this is important from the simulation perspective then I think it should also be relevant from the specification perspective. Nevertheless, you are an expert in this field so I trust your judgement :)

Reply:
While working on the simulation, now I think it’s better to have a param for the num of connections :) I’ll add it in this PR soon

Reply:
@madxor ed9b0ab I added the `peering_degree` param to the `NodeConfig`, so that it can be configured by each node. From its [definition](https://docs.libp2p.io/concepts/pubsub/overview/#full-message) in libp2p gossipsub, it means the **target** number of peers that a node should maintain. For the simplicity in this spec, we use it as the **maximum** number of peers that a node can maintain because I think we don't need to stick too much to libp2p in this spec. I think the spec needs to be generic and it's enough to have only the **maximum** number of peers as a parameter in this spec.

Reply:
Thanks! I agree also I think that we should treat the `peering_degree` as "expected" number of connections a node will have open. This is good enough for our simulation purposes as we can evaluate the properties of the system assuming unified level of connectivity for all nodes.

However, in real life a node might deviate from this and in current design we might not have any tool to prevent the node from opening to many or to little connections.

Reply:
True. In libp2p, this parameter means the 'expected' number of connections. I clarified the comment in the code: d3e8b02. This parameter will be used as the 'max' value in the simulation, for simplicity.

https://docs.libp2p.io/concepts/pubsub/overview/#full-message
> In libp2p’s default implementation the ideal network peering degree is 6 with anywhere from 4–12 being acceptable.

Reply:
After merging this PR to the master, I'll open subsequent PRs soon.